consensus/src/round_manager.rs

```diff
@@ -1107,18 +1105,47 @@ impl RoundManager {
                 return Ok(());
             }
 
-            if order_vote_msg.order_vote().ledger_info().round()
-                > self.block_store.sync_info().highest_ordered_round()
+            let highest_ordered_round = self.block_store.sync_info().highest_ordered_round();
+            let highest_ordered_epoch = self.block_store.sync_info().highest_ordered_cert().epoch();
+            let order_vote_round = order_vote_msg.order_vote().ledger_info().round();
+            let li_digest = order_vote_msg.order_vote().ledger_info().hash();
+            if order_vote_round > highest_ordered_round
+                && order_vote_round < highest_ordered_round + 100
+                && order_vote_msg.epoch() >= highest_ordered_epoch
```

Comment:
why we're checking epochs here? only same epoch message should be processed in round manager?

Reply:
I thought it avoid replay attacks when a validator replays messages from the above round range but from an older epoch.

Reply:
we should verify epoch in `verify_order_vote` function instead of here, I just realize we didn't check for that

Reply:
Actually, even for vote messages, we are not currently checking in the verify function that the vote's epoch matches with the local epoch. Is that intentional?

Reply:
oh I remember now, we check the epoch for every message before it enters the verification.
https://github.com/aptos-labs/aptos-core/blob/793eeb7273c25543a15ebd94de4db37335f0d8b2/consensus/src/epoch_manager.rs#L1503

Reply:
Gotcha. Removed the epoch check.